src/pages/calendar/components/Requisites.tsx

```diff
@@ -0,0 +1,192 @@
+                Earn a minimum grade of {req.grade} in each of the following:
+              </ListItem>
+            </UnorderedList>
+          )}
+          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}{' '}
+          {/* Ensure to increase the indentation level for nested elements */}
+        </Box>
+      );
+    }
+  } else if ('reqListItems' in req) {
+    const nestedReqs = nestedReq(req);
+    return <Box>{nestedReqs}</Box>; // Displays the list of requisites
+  } else if ('code' in req) {
+    // Get the current term and requisite details
+    const subject = req.subject;
+    const code = req.code;
+
+    // Get course details from the backend
+    return (
+      <GetCourseDetails term={currTerm} subject={req.subject} code={req.code}>
+        {(courseDetails, isLoading) => {
+          // If its loading then render a skeleton
+          if (isLoading.loading) {
+            return (
+              <UnorderedList>
+                <ListItem
+                  title={`${subject} ${code}`}
+                  style={{ marginLeft: `${indentationLevel * 40}px`, marginRight: `100px` }}
+                >
+                  <Skeleton>
+                    <Text>
+                      Loading course details of {req.subject} {req.code}
+                    </Text>
+                  </Skeleton>
+                </ListItem>
+              </UnorderedList>
+            );
+          } else if (courseDetails) {
+            // Extract course details and format credits
+            const pid = courseDetails.pid;
+            const credits = courseDetails.credits.credits;
+            const creditsVisual =
+              credits.min === credits.max ? `(${credits.max})` : `(${credits.min} - ${credits.max})`;
+
+            // Render course details with a hyperlink to requisite course page on courseup
+            return (
+              <UnorderedList>
+                <ListItem title={`${subject} ${code}`} style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                  <Text _hover={{ color: 'blue.600' }} color="blue.400" as="span">
+                    <Link to={`/calendar/${currTerm}/${subject}?pid=${pid}`}>{`${subject} ${code}`}</Link>
+                  </Text>
+                  {` - ${courseDetails.title} ${creditsVisual}`}
+                </ListItem>
+              </UnorderedList>
+            );
+          } else {
+            // Render default state
+            return (
+              <UnorderedList>
+                <ListItem
+                  title={`${subject} ${code}`}
+                  style={{ marginLeft: `${indentationLevel * 40}px` }}
+                >{`${subject} ${code}`}</ListItem>
+              </UnorderedList>
+            );
+          }
+        }}
+      </GetCourseDetails>
+    );
+  }
+  return <></>;
+}
+
+// Get nested reqs from NestedPreCoRequisites object
+export function nestedReq(req: NestedPreCoRequisites): Array<string | NestedPreCoRequisites | KualiCourse> {
```

Comment:
Not sure what the point of this being it's own function is. When it's used it could be
```ts
const nestedReqs = req.reqList ?? [];
```
Since it's used in multiple places, but within the same component (just in different if statements), it could just be defined at the top level, right?